### ISSUE TITLE
webpack --watch mode=development doesn't actually watch for changes

### ISSUE THREAD

**Issue by csush** [ucbdrive/scalabel#156]
I'm running the latest commit, but even in a ~1 month old commit webpack does not watch for changes in the entries. Moreover, in the latest commit the weback development command concludes instead of running continuously.

**Comment by noahjimenez5844** [comment]
Are you sure that your cache is disabled? If not, on chrome go to developer tools (ctrl + shift + j for windows), and click the disable cache option in the network tab.

**Comment by csush** [comment]
> Are you sure that your cache is disabled? If not, on chrome go to developer tools (ctrl + shift + j for windows), and click the disable cache option in the network tab.

@noahjimenez5844  did that. Now the webpack development command runs continuously but there is still no auto-reloading.

**Comment by noahjimenez5844** [comment]
@csush By auto-reloading do you mean recompiling the changed source code or refreshing the browser? If you mean the latter, unfortunately this is not implemented, and you have to manually refresh you browser (and it can take a few seconds to recompile). If your code isn't re-compiling, can you post your webpack logs?

**Comment by csush** [comment]
> @csush By auto-reloading do you mean recompiling the changed source code or refreshing the browser? If you mean the latter, unfortunately this is not implemented, and you have to manually refresh your browser (and it can take a few seconds to recompile). If your code isn't re-compiling, can you post your webpack logs?

My bad, wasn't clear earlier. It's not re-compiling. It's an unnecessary overhead to go to the terminal and enter the command every time I make changes. I don't mind refreshing the browser.

Here's the log:
`$ node_modules/.bin/npx webpack --watch --config webpack.config.js --mode=development

webpack is watching the files…

ℹ ｢atl｣: Using typescript@3.5.2 from typescript
ℹ ｢atl｣: Using tsconfig.json from /dir/to/scalabel/tsconfig.json
ℹ ｢atl｣: Checking started in a separate process...
ℹ ｢atl｣: Time: 282ms
Hash: 499f0ce07230be9f7023
Version: webpack 4.35.2
Time: 15649ms
Built at: 07/18/2019 11:17:30 AM
                         Asset       Size       Chunks                    Chunk Names
      ../annotation/image.html     15 KiB               [emitted]         
      ../annotation/label.html  230 bytes               [emitted]         
../annotation/point_cloud.html   7.16 KiB               [emitted]         
         ../control/admin.html  245 bytes               [emitted]         
        ../control/create.html   11.1 KiB               [emitted]         
     ../control/dashboard.html   9.29 KiB               [emitted]         
         ../control/login.html   5.17 KiB               [emitted]         
        ../control/vendor.html    8.3 KiB               [emitted]         
        ../control/worker.html  247 bytes               [emitted]         
            ../css/general.css   1.24 KiB               [emitted]         
              ../css/image.css   8.55 KiB               [emitted]         
              ../css/style.css   3.69 KiB               [emitted]         
        ../dev/speed_test.html   1.64 KiB               [emitted]         
                ../favicon.ico   4.19 KiB               [emitted]         
            ../img/loading.png   5.48 KiB               [emitted]         
     ../img/scalabel_white.svg   3.92 KiB               [emitted]         
                 ../index.html   2.88 KiB               [emitted]         
                      admin.js   1.63 MiB        admin  [emitted]  [big]  admin
                  admin.js.map   1.62 MiB        admin  [emitted]         admin
                     create.js    320 KiB       create  [emitted]  [big]  create
                 create.js.map    415 KiB       create  [emitted]         create
                  dashboard.js    289 KiB    dashboard  [emitted]  [big]  dashboard
              dashboard.js.map    375 KiB    dashboard  [emitted]         dashboard
                      image.js    878 KiB        image  [emitted]  [big]  image
                  image.js.map   1.14 MiB        image  [emitted]         image
                   image_v2.js   1.39 MiB     image_v2  [emitted]  [big]  image_v2
               image_v2.js.map   1.65 MiB     image_v2  [emitted]         image_v2
                      label.js   5.71 MiB        label  [emitted]  [big]  label
                  label.js.map   5.88 MiB        label  [emitted]         label
                point_cloud.js   1.87 MiB  point_cloud  [emitted]  [big]  point_cloud
            point_cloud.js.map   2.38 MiB  point_cloud  [emitted]         point_cloud
                 speed_test.js   22.2 KiB   speed_test  [emitted]         speed_test
             speed_test.js.map   31.3 KiB   speed_test  [emitted]         speed_test
                     vendor.js    279 KiB       vendor  [emitted]  [big]  vendor
                 vendor.js.map    363 KiB       vendor  [emitted]         vendor
                     worker.js   1.48 MiB       worker  [emitted]  [big]  worker
                 worker.js.map    1.5 MiB       worker  [emitted]         worker
Entrypoint create [big] = create.js create.js.map
Entrypoint image [big] = image.js image.js.map
Entrypoint image_v2 [big] = image_v2.js image_v2.js.map
Entrypoint worker [big] = worker.js worker.js.map
Entrypoint admin [big] = admin.js admin.js.map
Entrypoint point_cloud [big] = point_cloud.js point_cloud.js.map
Entrypoint speed_test = speed_test.js speed_test.js.map
Entrypoint dashboard [big] = dashboard.js dashboard.js.map
Entrypoint vendor [big] = vendor.js vendor.js.map
Entrypoint label [big] = label.js label.js.map
[./app/src/js/common/session_init.tsx] 1.42 KiB {label} [built]
[./app/src/js/common/session_single_init.js] 4 KiB {image_v2} [built]
[./app/src/js/dev/speed_test.js] 8.33 KiB {speed_test} [built]
[./app/src/js/entries/admin.tsx] 205 bytes {admin} [built]
[./app/src/js/entries/image.index.js] 1.4 KiB {image_v2} [built]
[./app/src/js/entries/label.index.ts] 89 bytes {label} [built]
[./app/src/js/entries/worker.tsx] 207 bytes {worker} [built]
[./app/src/js/v1/box2d.js] 14.4 KiB {image} [built]
[./app/src/js/v1/create.js] 6.03 KiB {create} [built]
[./app/src/js/v1/dashboard.index.js] 1.07 KiB {dashboard} [built]
[./app/src/js/v1/image.index.js] 2.08 KiB {image} [built]
[./app/src/js/v1/image.js] 54.5 KiB {image} [built]
[./app/src/js/v1/point_cloud/box3d.js] 8.89 KiB {point_cloud} [built]
[./app/src/js/v1/point_cloud/point_cloud.index.js] 445 bytes {point_cloud} [built]
[./app/src/js/v1/vendor.index.js] 395 bytes {vendor} [built]
    + 614 hidden modules`

This log is from a month old commit of the original project, I have made some changes of my own. But re-compiling automatically has never worked now that I look back. I have cloned the latest commit as well separately and it doesn't re-compile on that either.

**Comment by noahjimenez5844** [comment]
@csush Is this the entire log? After you make a change and save, do you get a message starting with "Checking started in a separate process..."? If not, that's pretty weird, and might have to do with how your editor is caching your code. For example with WebStorm, disable **Settings -> System Settings -> "safe write"**, as suggested [here](https://stackoverflow.com/questions/26708205/webpack-watch-isnt-compiling-changed-files) (and also maybe try the other suggested solutions, such as increasing the number of watchers).  Webpack is working for me with both the current commit and previous ones as well using 4.35.2.

**Comment by csush** [comment]
@noahjimenez5844 Yes, thats the entire log. It was not updating when I modified any of the watched files. But now it is! I used one of the suggestions in the [post](https://stackoverflow.com/a/45655104) you referred to.

Adding this in the config did it for me:
`
watchOptions: {
  poll: true,
  ignored: /node_modules/
}
`

From the response: 

> Polling is the continuous checking of other programs or devices by one program or device to see what state they are in, usually to see whether they are still connected or want to communicate. So setting poll: true allows webpack to check the state of your program to see if any changes have been made, or at least that what I assume is happening.

Although now I'm not sure why just doing a --watch did not work. I assumed these values to be default, otherwise it makes no sense to have the --watch option.

Thanks a lot for the help!
